astropy/io/votable/tree.py

```diff
@@ -3089,6 +3098,72 @@ def _parse_fits(self, iterator, extnum, config):
 
         return array
 
+    def _parse_parquet(self, iterator, config):
+        """
+        Functionality to parse parquet files that are embedded
+        in VOTables.
+        """
+        # looks like votable already has a "Table" imported.
+        from astropy.table import Table as Table2
+
+        for start, tag, data, pos in iterator:
+            if tag == "STREAM":
+                if start:
+                    warn_unknown_attrs(
+                        "STREAM",
+                        data.keys(),
+                        config,
+                        pos,
+                        ["type", "href", "actuate", "encoding", "expires", "rights"],
+                    )
+                    href = data["href"]
+                    encoding = data.get("encoding", None)
+                else:
+                    break
+
+            else:  # in this case, there is no STREAM, hence no file linked.
+                href = ""
+
+        if not href.startswith(("http", "ftp", "file")):
+            vo_raise(
+                "The vo package only supports remote data through http, ftp or file",
+                self._config,
+                self._pos,
+                NotImplementedError,
+            )
+
+        # Hack to keep windows working
+        try:
+            fd = urllib.request.urlopen(href)
+        except urllib.error.URLError:
+            # Hack to keep windows working
+            if href.startswith("file://"):
+                fd = urllib.request.urlopen(f"file:{href[7:]}")
+            # Relative path to parquet part should be relative from the votable
+            elif href.startswith("file:"):
+                parquet = os.path.join(os.path.dirname(config["filename"]), href[5:])
+                fd = urllib.request.urlopen(f"file:{parquet}")
+
+        if encoding is not None:
+            if encoding == "gzip":
+                fd = gzip.GzipFile(href, "r", fileobj=fd)
```

Comment:
Shouldn't the GzipFile be closed? I had to close it here and wonder why you are able to get away with it:

* https://github.com/astropy/astropy/pull/15359

Reply:
I don't exactly know why it works either, but my bet would be because L3160 picks up the same fd again and closes it properly.

Reply:
I cannot find where it is actually closed. I got lost in the `parse` logic.

Reply:
Table.read() should ideally close stuff, right? It certainly cannot be closed in this line here, before it hits that read line below.